web/src/pages/api/collection/all.ts

```diff
@@ -0,0 +1,45 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { prisma } from "../../../server/db/client";
+
+interface Request extends NextApiRequest {
+  body: {
+    userId: string;
```

Comment:
We shouldn't be expecting to receive body for a GET endpoint (which I think it is). Instead, we should grab these from the query parameters.

I would recommend we add a field searchUserId or getUserId. Currently we are using userId to verify that you are the user you say you are when validating endpoints. If we want to grab some data for another userId then we have to add a new one.